### ISSUE TITLE
Can't focus input element in modal

### ISSUE THREAD

**Issue by asauty** [NG-ZORRO/ng-zorro-antd#7257]
### Reproduction link
[https://stackblitz.com/edit/angular-ykkcar](https://stackblitz.com/edit/angular-ykkcar)

### Steps to reproduce
I 'm trying to focus an input from a component passed to the content of a modal.
At the lifecycle hook AfterViewInit, I focus the nativeElement of input in the TestComponent.
But it seems the focus is given to an another element.
With a timeout of 1sec, my input is focused. But this is not the solution I expect.

### What is expected?
The input element should be focus.

### What is actually happening?
The input element isn't focus.

| Environment | Info |
|---|---|
| ng-zorro-antd | 13.0.1 |
| Browser | Opera |

**Comment by chenc041** [comment]
try this 
```ts
setTimeout(() => this.input.nativeElement.focus(), 0);
```

**Comment by cyfrin** [comment]
> try this
> 
> ```ts
> public ngAfterViewInit(): void {
>     // this.input.nativeElement.focus();
>     setTimeout(() => this.input.nativeElement.focus(), 0);
>   }
> ```

This thing works like a charm as Modal, bus when I try to use the same component as Drawer Component is does not work.
Any work around will be deeply appreciated??

Stackblitz
[https://stackblitz.com/edit/angular-ykkcar-slrnnw]

**Comment by asauty** [comment]
It's functional, but it's not maintainable. 

> try this
> 
> ```ts
> public ngAfterViewInit(): void {
>     // this.input.nativeElement.focus();
>     setTimeout(() => this.input.nativeElement.focus(), 0);
>   }
> ```

**Comment by chenc041** [comment]
> It's functional, but it's not maintainable.
> 
> > try this
> > ```ts
> > public ngAfterViewInit(): void {
> >     // this.input.nativeElement.focus();
> >     setTimeout(() => this.input.nativeElement.focus(), 0);
> >   }
> > ```

The lifecycle method works fine, but angular calls it so frequently that I don't think it's worth it
```ts
export class TestComponent implements AfterViewChecked {
  @ViewChild('test')
  public input!: ElementRef<HTMLElement>;

  public ngAfterViewChecked(): void {
    console.log(this.input.nativeElement, '1111');
    this.input.nativeElement.focus();

    // setTimeout(() => this.input.nativeElement.focus(), 1000);
  }
}
```

**Comment by chenc041** [comment]
> > try this
> > ```ts
> > public ngAfterViewInit(): void {
> >     // this.input.nativeElement.focus();
> >     setTimeout(() => this.input.nativeElement.focus(), 0);
> >   }
> > ```
> 
> This thing works like a charm as Modal, bus when I try to use the same component as Drawer Component is does not work. Any work around will be deeply appreciated??
> 
> Stackblitz [https://stackblitz.com/edit/angular-ykkcar-slrnnw]

try change 0 -> 10 ....
but why it work, I do not know 😂

**Comment by cyfrin** [comment]
Yes, it may be be expecting some delay, and thanks for the workaround.
But why it is not working with with the nz-select

`<nz-select nz-input formControlName="prefixID" nzPlaceHolder="Mr" #test>
    <nz-option nzValue="0" nzLabel="None"></nz-option>
  </nz-select>`

Throwing this error
`core.mjs:6449 ERROR TypeError: Cannot read properties of undefined (reading 'focus')`

**Comment by chenc041** [comment]
> Yes, it may be be expecting some delay, and thanks for the workaround. But why it is not working with with the nz-select
> 
> ```
> <nz-select nz-input formControlName="title" nzPlaceHolder="Mr" #test>
>   <nz-option nzValue="0" nzLabel="None"></nz-option>
> </nz-select>
> ```
> 
> Throwing this error `core.mjs:6449 ERROR TypeError: Cannot read properties of undefined (reading 'focus')`

try this
```ts
import {
  AfterViewInit,
  ChangeDetectionStrategy,
  Component,
  ViewChild,
} from '@angular/core';
import { NzSelectComponent } from 'ng-zorro-antd/select';

@Component({
  selector: 'app-demo',
  template: `
    <nz-select
      #select
      nzShowSearch
      nzAllowClear
      nzPlaceHolder="Select a person"
      [(ngModel)]="selectedValue"
    >
      <nz-option nzLabel="Jack" nzValue="jack"></nz-option>
      <nz-option nzLabel="Lucy" nzValue="lucy"></nz-option>
      <nz-option nzLabel="Tom" nzValue="tom"></nz-option>
    </nz-select>
  `,
  styleUrls: ['./demo.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class DemoComponent implements AfterViewInit {
  selectedValue = null;

  @ViewChild('select') select!: NzSelectComponent;

  ngAfterViewInit() {
    this.select.focus();
  }
}

```

**Comment by cyfrin** [comment]
Thanks so lot is works with a timeout as with case of input

```ts
setTimeout(() => this.select.focus(), 2);
```

**Comment by cyfrin** [comment]
```ts
import { Component } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { NzModalService, NzModalRef } from 'ng-zorro-antd/modal';

import { NzMessageService } from 'ng-zorro-antd/message';

@Component({
  selector: 'nz-demo-modal-info',
  template: `
  <button nz-button nzType="primary" nzDanger (click)="deleteModal()">Async Http Call</button>
  `,
  styles: [
    `
      button {
        margin-right: 8px;
      }
    `,
  ],
})
export class NzDemoModalInfoComponent {
  constructor(
    private modal: NzModalService,
    private http: HttpClient,
    private message: NzMessageService
  ) {}

  deleteModal(): void {
    const modal: NzModalRef = this.modal.create({
      nzTitle: 'Delete Confirm',
      nzContent: 'Are you sure to delete?',
      nzFooter: [
        {
          label: 'Close',
          type: 'text',
          onClick: () => modal.destroy(),
        },
        {
          label: 'Delete',
          type: 'primary',
          danger: true,
          loading: false,
          onClick: () => {
            // loading not working
            // this.loading = true;
            this.http
              .delete('https://jsonplaceholder.typicode.com/postss/1')
              .subscribe({
                next: (res) => {
                  console.log(res, 'Deleted');
                  this.message.create('success', 'Successfully Deleted');
                  modal.destroy();
                  // loading not working
                  // this.loading = false;
                },
                error: (err) => {
                  console.log(err, 'Deletion Error');
                  this.message.create(
                    'error',
                    'Delete Failed, please try again'
                  );
                  // loading not working
                  // this.loading = false;
                },
              });
          },
        },
      ],
    });
  }
}
```
@chenc041 can you please help me how to change the state of a custom button on modal to loading during a async call.
[https://stackblitz.com/edit/angular-ykkcar-3tny1q](https://stackblitz.com/edit/angular-ykkcar-3tny1q)

**Comment by chenc041** [comment]
> ```ts
> import { Component } from '@angular/core';
> import { HttpClient } from '@angular/common/http';
> import { NzModalService, NzModalRef } from 'ng-zorro-antd/modal';
> 
> import { NzMessageService } from 'ng-zorro-antd/message';
> 
> @Component({
>   selector: 'nz-demo-modal-info',
>   template: `
>   <button nz-button nzType="primary" nzDanger (click)="deleteModal()">Async Http Call</button>
>   `,
>   styles: [
>     `
>       button {
>         margin-right: 8px;
>       }
>     `,
>   ],
> })
> export class NzDemoModalInfoComponent {
>   constructor(
>     private modal: NzModalService,
>     private http: HttpClient,
>     private message: NzMessageService
>   ) {}
> 
>   deleteModal(): void {
>     const modal: NzModalRef = this.modal.create({
>       nzTitle: 'Delete Confirm',
>       nzContent: 'Are you sure to delete?',
>       nzFooter: [
>         {
>           label: 'Close',
>           type: 'text',
>           onClick: () => modal.destroy(),
>         },
>         {
>           label: 'Delete',
>           type: 'primary',
>           danger: true,
>           loading: false,
>           onClick: () => {
>             // loading not working
>             // this.loading = true;
>             this.http
>               .delete('https://jsonplaceholder.typicode.com/postss/1')
>               .subscribe({
>                 next: (res) => {
>                   console.log(res, 'Deleted');
>                   this.message.create('success', 'Successfully Deleted');
>                   modal.destroy();
>                   // loading not working
>                   // this.loading = false;
>                 },
>                 error: (err) => {
>                   console.log(err, 'Deletion Error');
>                   this.message.create(
>                     'error',
>                     'Delete Failed, please try again'
>                   );
>                   // loading not working
>                   // this.loading = false;
>                 },
>               });
>           },
>         },
>       ],
>     });
>   }
> }
> ```
> 
> @chenc041 can you please help me how to change the state of a custom button on modal to loading during a async call. https://stackblitz.com/edit/angular-ykkcar-3tny1q

try this~
```ts
deleteModal(): void {
    const modal: NzModalRef = this.modal.create({
      nzTitle: 'Delete Confirm',
      nzContent: 'Are you sure to delete?',
      nzFooter: [
        {
          label: 'Close',
          type: 'text',
          onClick: () => modal.destroy(),
        },
        {
          label: 'Delete',
          type: 'primary',
          danger: true,
          loading: () => this.isLoading,
          onClick: () => {
            // loading not working
            // this.loading = true;
            this.isLoading = true;
            this.http
              .delete('https://jsonplaceholder.typicode.com/postss/1')
              .subscribe({
                next: (res) => {
                  this.isLoading = false;
                  this.message.create('success', 'Successfully Deleted');
                  modal.destroy();
                  // loading not working
                  // this.loading = false;
                },
                error: (err) => {
                  this.isLoading = false;
                  console.log(err, 'Deletion Error');
                  this.message.create(
                    'error',
                    'Delete Failed, please try again'
                  );
                  // loading not working
                  // this.loading = false;
                },
              });
          },
        },
      ],
    });
  }
```

**Comment by cyfrin** [comment]
😬 silly me, I tried earlier with `loading: this.isDeleting` but din't work but with `loading: () => this.isDeleting`,  is it working as expected.

**Comment by chenc041** [comment]
> 

haha 😂
any other questions? and if has other questions, you would better open a new issue
because this issue will disturb others

**Comment by chenc041** [comment]
> 😬 silly me, I tried earlier with `loading: this.isDeleting` but din't work but with `loading: () => this.isDeleting`, is it working as expected.

haha 😂
any other questions? and if has other questions, you would better open a new issue
because this issue will disturb others

**Comment by cyfrin** [comment]
Yeah, I really thought of opening a new issue earlier today, but somehow posted here, I'm thinking of cleaning up the unrelated comments
